src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ExecuteActionTests.cs

```diff
@@ -0,0 +1,169 @@
+
+            try
+            {
+                outputDirectory = Directory.CreateDirectory(Path.Combine(Path.GetTempPath(), "ExecuteAction", Guid.NewGuid().ToString()));
+                string textFileOutputPath = Path.Combine(outputDirectory.FullName, "file.txt");
+
+                const string testMessage = "TestMessage";
+
+                options.Path = DotNetHost.HostExePath;
+                options.Arguments = GenerateArgumentsString(new string[] { "TextFileOutput", textFileOutputPath, testMessage });
+
+                using CancellationTokenSource cancellationTokenSource = new CancellationTokenSource(TokenTimeoutMs);
+
+                CollectionRuleActionResult result = await action.ExecuteAsync(options, null, cancellationTokenSource.Token);
+
+                ValidateActionResult(result, "0");
+
+                Assert.Equal(testMessage, File.ReadAllText(textFileOutputPath));
+            }
+            finally
+            {
+                try
+                {
+                    outputDirectory?.Delete(recursive: true);
+                }
+                catch
+                {
+                }
+            }
+        }
+
+        [Fact]
+        public async Task ExecuteAction_InvalidPath()
+        {
+            ExecuteAction action = new();
+
+            ExecuteOptions options = new();
+
+            string uniquePathName = Guid.NewGuid().ToString();
+
+            options.Path = uniquePathName;
+            options.Arguments = GenerateArgumentsString(Array.Empty<string>());
+
+            using CancellationTokenSource cancellationTokenSource = new CancellationTokenSource(TokenTimeoutMs);
+
+            FileNotFoundException fileNotFoundException = await Assert.ThrowsAsync<FileNotFoundException>(
+                () => action.ExecuteAsync(options, null, cancellationTokenSource.Token));
+
+            Assert.Equal(string.Format(Strings.ErrorMessage_FileNotFound, uniquePathName), fileNotFoundException.Message);
+        }
+
+        [Fact]
+        public async Task ExecuteAction_IgnoreExitCode()
+        {
+            ExecuteAction action = new();
+
+            ExecuteOptions options = new();
+
+            options.Path = DotNetHost.HostExePath;
+            options.Arguments = GenerateArgumentsString(new string[] { "NonzeroExitCode" });
+            options.IgnoreExitCode = true;
+
+            using CancellationTokenSource cancellationTokenSource = new CancellationTokenSource(TokenTimeoutMs);
+
+            CollectionRuleActionResult result = await action.ExecuteAsync(options, null, cancellationTokenSource.Token);
+
+            ValidateActionResult(result, "1");
+        }
+
+        private static string GenerateArgumentsString(string[] additionalArgs)
+        {
+            Assembly currAssembly = Assembly.GetExecutingAssembly();
+
+            return AssemblyHelper.GetAssemblyArtifactBinPath(currAssembly, "Microsoft.Diagnostics.Monitoring.ExecuteActionApp", TargetFrameworkMoniker.NetCoreApp31)
```

Comment:
nit: It might read better (and be easier to extend in the future) if this was something more like:

```cs
List<string> args = new();

// Entrypoint assembly
args.Add(AssemblyHelper.GetAssemblyArtifactBinPath(currAssembly, "Microsoft.Diagnostics.Monitoring.ExecuteActionApp", TargetFrameworkMoniker.NetCoreApp31));
// Entrypoint arguments
args.AddRange(additionalArgs)

return string.Join(' ', args);
```